### DISCUSSION TITLE
Data importer timeout connection

### DISCUSSION THREAD

**Discussion by Rebislori** [firefly-iii/firefly-iii#8364] [Firefly III] [answered]
Hi everyone, sorry for being such a noob.

I installed firefly iii in docker-compose under proxmox.
Then i installed data importer.

They both run, but i'm not able to connect data importer to firefly-core.
When i enter the client ID and click to connect i get a connection timeout error.

In importer.env i set the internal ip of my firefly container (my first error was setting it to the external address).

I really don't know that could help you helping me, the logs look fine.

Could you please help me troubleshooting this situation?

Thanks in advance.

**Comment by JC5** [comment]
Tell me, when it times out. What is the address in the address bar of the browser? Is it the data importer, or is it trying to connect to Firefly III?

**Comment by Rebislori** [comment]
Hi, this morning i fiddled around and redid everything from scratch.
Now it's better, i get to the authorization request page.

After clicking on "authorize" i get the "whoops! 500 :(" message.
And this is the URL (192.168.15.29:8082 is the url i use to reach data importer in my browser):
http://192.168.15.29:8082/callback?code=def502002969e7a661afd90087899cc1441b7643cc668f92a27afec49378a477ac7471a0233eae4aa9899408c3587be912158e57b157745c86453a263e3f46299153f71aadfc502e5db4a9df76e13f5a1f0e65f122f7f1c23d1cf875733c97545d6de45d00d9276fb392e707c62eed7e99979b31495d0d1d636f60f26bfb59a5249c0744123aaecebc5a5d56bb817a49fd175347e79a339fa63e319a2c9c0dfa0f40dd7d2a338b027e9dad1581e8ca7dd2e8c3ebab19f77ce10c94ac69b98be13fc886b5a3161d26bcde5721000b00e9fae68413ee53f53b3e5441d3e954f6127c6f4cb429fbc3dc68373419afdf884229bcc8edb5617c65c398221edbbb3c204a33d3352abbcd9b90c5388284510a0910e1519fbd8e58390f2450749002dde9c739b7748316a37b5632a47236240779539daab7930ea5b4841d71f20b6777d453adf5ce8a2973ab18c3efb3a904b43c66809a29210121d9967947d1ccfab5a704647ea20caa290abf467092294e39d5e7f5e806a7fd23b6aa92148cf2b1307a34a7f6230625ee8fa6addb88d6ac&state=D1MyEaB8PZRt3jDlCeNuLxaCPlz4rNU8tndnQ0gg

One thing i don't get is: i created the container in docker-compose, and now i can manage them in portainer. 
But when i change something in .env files, those changes are not reflected in portainer even after recreating the container; i have to edit them inside portainer and then recreate it. is this normal?

**Comment by Rebislori** [comment, marked as answer]
Well.....

![immagine](https://github.com/firefly-iii/firefly-iii/assets/118441237/ef67af8f-6286-4248-b77c-e3c72efa1650)

that was it. wrong ip!

**Reply by JC5** [reply]
Nicely spotted!
